test/skills/core.py

```diff
@@ -0,0 +1,99 @@
+import unittest
+from os.path import join, dirname
+
+from mycroft.skills.core import load_regex_from_file, load_regex
+from mycroft.util.log import getLogger
+
+__author__ = 'eward'
+logger = getLogger(__name__)
+
+
+class MockEmitter(object):
+    def __init__(self):
+        self.types = []
+        self.results = []
+
+    def emit(self, message):
+        self.types.append(message.message_type)
+        self.results.append(message.metadata['regex'])
+
+    def get_types(self):
+        return self.types
+
+    def get_results(self):
+        return self.results
+
+    def reset(self):
+        self.types = []
+        self.results = []
+
+
+class MycroftSkillTest(unittest.TestCase):
+    emitter = MockEmitter()
+
+    def check_load_from_file(self, path, emitter, regex_list):
+        load_regex_from_file(path, emitter)
+        self.check_emitter(emitter, regex_list)
+
+    def check_load(self, path, emitter, regex_list):
+        load_regex(path, emitter)
+        self.check_emitter(emitter, regex_list)
+
+    def check_emitter(self, emitter, regex_list):
```

Comment:
`emitter` is available from `self` you don't need to pass it all way down here to use it. Just use `self.emitter` inside this method and remove `emitter` from this method signature. As a result, will also remove all `emitter` from the other methods signatures and calls. In the end, your code will be much simpler.